e2e/nomostest/config_sync.go

```diff
@@ -873,118 +855,121 @@ func RepoSyncObjectV1Beta1FromOtherRootRepo(nt *NT, nn types.NamespacedName, rep
 
 // setupCentralizedControl is a pure central-control mode.
 // A default root repo (root-sync) manages all other root repos and namespace repos.
-func setupCentralizedControl(nt *NT, opts *ntopts.New) {
+func setupCentralizedControl(nt *NT, reconcileTimeout *time.Duration) {
 	nt.T.Log("[SETUP] Centralized control")
 
 	rsCount := 0
 
 	// Add any RootSyncs specified by the test options
-	for rsName := range opts.RootRepos {
+	for rsName := range nt.RootRepos {
 		// The default RootSync is created manually, don't check it in the repo.
 		if rsName == configsync.RootSyncName {
 			continue
 		}
 		rs := RootSyncObjectV1Beta1FromRootRepo(nt, rsName)
-		if opts.MultiRepo.ReconcileTimeout != nil {
-			rs.Spec.SafeOverride().ReconcileTimeout = toMetav1Duration(*opts.MultiRepo.ReconcileTimeout)
+		if reconcileTimeout != nil {
+			rs.Spec.SafeOverride().ReconcileTimeout = toMetav1Duration(*reconcileTimeout)
 		}
 		nt.RootRepos[configsync.RootSyncName].Add(fmt.Sprintf("acme/namespaces/%s/%s.yaml", configsync.ControllerNamespace, rsName), rs)
+		nt.rootRepoObjects[configsync.RootSyncName][core.IDOf(rs)] = true
 		nt.RootRepos[configsync.RootSyncName].CommitAndPush("Adding RootSync: " + rsName)
 	}
 
-	if len(opts.NamespaceRepos) == 0 {
-		// Wait for the RootSyncs and exit early
-		if err := nt.WatchForAllSyncs(); err != nil {
-			nt.T.Fatal(err)
-		}
-		return
-	}
-
-	cr := nt.RepoSyncClusterRole()
-	nt.RootRepos[configsync.RootSyncName].Add("acme/cluster/cr.yaml", cr)
+	if len(nt.NonRootRepos) > 0 {
```

Comment:
refactoring this `if` statement makes this diff unnecessarily large imo. also makes the `if` block quite long which makes it harder to follow

Reply:
That's why i had removed the if statement in a previous PR, but then that broke metric validation, which was being done after the If block.

Reply:
hypothetically, but it's possible they may diverge in the future. This is just some bits extracted from https://github.com/GoogleContainerTools/kpt-config-sync/pull/412, which changes all the metrics validation impls. I just wanted them to be consistent first.

Reply:
done